src/main/java/net/spy/memcached/TLSConnectionHandler.java

```
@@ -0,0 +1,345 @@
+  * @param timeoutInMillis the amount of time to wait for TLS handshake finish before timeout.
+  * @return true if handshake is successful, false if handshake is unsuccessful.
+  * @throws IOException if there is an error occurred during read/write to the socket channel.
+  * @throws SSLException if a problem was encountered while signaling the SSLEngine to begin a new handshake
+  *           or if a problem was encountered while processing the data that caused the SSLEngine to abort.
+  * @throws IllegalStateException if the client/server mode has not yet been set.
+  * @throws IllegalArgumentException if the handshake status is invalid.
+  * @throws OperationTimeoutException if global operation timeout is exceeded
+  */ 
+  public boolean doTlsHandshake(long timeoutInMillis) throws IOException {
+
+    // Create a new SSLEngine session
+    SSLSession session = sslEngine.getSession();
+      
+    // Get the current size of largest SSL/TLS packet that is expected when using this session.
+    int packetBufferSize = session.getPacketBufferSize();
+    myNetData = ByteBuffer.allocate(packetBufferSize);
+    peerNetData = ByteBuffer.allocate(packetBufferSize);
+    
+    // Get the current size of largest SSL/TLS application that is expected when using this session.
+    int appBufferSize = session.getApplicationBufferSize();
+    myAppData = ByteBuffer.allocate(appBufferSize);
+    peerAppData = ByteBuffer.allocate(appBufferSize);
+
+    // Prepare buffer for use
+    myNetData.clear();
+    peerNetData.clear();
+    myAppData.clear();
+    peerAppData.clear();
+
+    // Set the start time before starting TLS handshake.
+    long startTimeDoTlsHandshake = System.nanoTime();
+    
+    try {
+      sslEngine.beginHandshake();
+    } catch (SSLException sslException) {
+      log.error("A problem was encountered while signaling the SSLEngine to begin a new handshake", sslException);
+    } catch (IllegalStateException e){
```

Do you really want to silently swallow those exceptions? Probably just don't handle them and let them bubble up if you can't begin a handshake for whatever reason?